pkg/storage/utils/decomposedfs/grants.go

```diff
@@ -240,18 +252,30 @@ func (fs *Decomposedfs) storeGrant(ctx context.Context, n *node.Node, g *provide
 		return err
 	}
 
+	var spaceType string
+	spaceGrant := ctx.Value(utils.SpaceGrant)
+	// this is not a grant on a space root we are just adding a share
+	if spaceGrant == nil {
+		spaceType = spaceTypeShare
+	}
+	// this is a grant to a space root, the receiver needs the space type to update the indexes
+	if sg, ok := spaceGrant.(struct{ SpaceType string }); ok && sg.SpaceType != "" {
+		spaceType = sg.SpaceType
+	}
+
+	// set the grant
```

Comment:
At this point `spaceType` could be an empty string if both conditions failed and looking at the `updateIndexes` function it doesn't seem to be a valid case. Do we have to bail out here in that case?

Reply:
If we have no space type then the updateIndexes can only create the `by-user-id` index. 
So IMO it can handle empty strings for the spaceTyoe

Reply:
https://github.com/micbar/reva/blob/5ae2aa5da34f2092a36ccef4703142fd2a1e5599/pkg/storage/utils/decomposedfs/spaces.go#L654